Log is undefined when plugin is added programatically

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

## Bug Report

**Error Description**
If plugin is added via `serverless.pluginManager.addPlugin` function as part of an other plugin `Globals.v3Utils.log` becomes `undefined` 

We have an internal plugin that looks like this (some un related parts removed for clarity):
```ts
import Serverless from 'serverless';
import Plugin, { PluginStatic } from 'serverless/classes/Plugin';
import { ServerlessCustomDomain } from 'serverless-domain-manager';

class DomainManagerPlugin implements Plugin {
  serverless: Serverless;

  hooks: Plugin.Hooks;

  constructor(serverless: Serverless) {
    this.serverless = serverless;
    this.hooks = {
      initialize: () => this.injectConfig(), // run in hook in case of config needing to be resolved, e.g. shard from params
    };

    // Does not work, need to add the plugin manually, package internal logger breaks for an unknown reason
    serverless.pluginManager.addPlugin(ServerlessCustomDomain as PluginStatic);
  }

  private injectConfig() {
    if (!this.serverless.service.custom) {
      this.serverless.service.custom = {};
    }
    this.serverless.service.custom.customDomain = {
      domainName: 'api.our.domain.com',
      basePath: 'our-base-path',
      stage: '$default',
      apiType: 'http',
      endpointType: 'regional',
      createRoute53Record: false,
      createRoute53IPv6Record: false,
    };
  }
}
```
If I remove the `serverless.pluginManager.addPlugin(...)` line and add `serverless-domain-manager` to the plugin section of serverless.yml everything works as expected 

**Command Run**
`sls deploy`

**Console Output**
```sh
Stack **** failed to deploy (60s)
Environment: linux, node 18.18.2, framework 3.37.0 (local), plugin 7.2.0, SDK 4.5.1
Credentials: Local, "default" profile
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
TypeError: Cannot read properties of undefined (reading 'verbose')
    at Logging.logInfo (****/node_modules/.pnpm/serverless-domain-manager@7.1.2_serverless@3.37.0/node_modules/serverless-domain-manager/dist/src/logging.js:24:43)
    at ServerlessCustomDomain.<anonymous> (****/node_modules/.pnpm/serverless-domain-manager@7.1.2_serverless@3.37.0/node_modules/serverless-domain-manager/dist/src/index.js:180:35)
    at Generator.throw (<anonymous>)
    at rejected (****/node_modules/.pnpm/serverless-domain-manager@7.1.2_serverless@3.37.0/node_modules/serverless-domain-manager/dist/src/index.js:6:65)
    
Error: Process completed with exit code 1.
```

**Domain Manager Configuration**
Replace this with your own `serverless.yml` file (anonymized, of course) to help us better resolve your issue.
```yaml
custom:
  customDomain:
      domainName: 'api.our.domain.com',
      basePath: 'our-base-path',
      stage: '$default',
      apiType: 'http',
      endpointType: 'regional',
      createRoute53Record: false,
      createRoute53IPv6Record: false,
```

**Versions**
- Domain Manager version(s): 7.1.2
- Node/npm version: 18.18.2
- Serverless Version: 3.37.0
- Lambda Code TypeScript/JavaScript

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [7.2.1] - 2023-11-28
+
+### Fixed 
+- Fixed log is undefined when plugin is added programatically
+- Updated packages
+
 ## [7.2.0] - 2023-11-24
 
 ### Fixed 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,7 +4,7 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
-## [7.2.1] - 2023-11-28
+## [7.2.1] - 2023-11-27
 
 ### Fixed 
 - Fixed log is undefined when plugin is added programatically

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "serverless-domain-manager",
-  "version": "7.2.0",
+  "version": "7.2.1",
   "engines": {
     "node": ">=14"
   },
@@ -50,7 +50,7 @@
   },
   "devDependencies": {
     "@types/mocha": "^10.0.6",
-    "@types/node": "^20.9.4",
+    "@types/node": "^20.10.0",
     "@types/shelljs": "^0.8.15",
     "aws-sdk-client-mock": "^3.0.0",
     "chai": "^4.3.10",
@@ -59,7 +59,7 @@
     "mocha-param": "^2.0.1",
     "nyc": "^15.1.0",
     "randomstring": "^1.3.0",
-    "serverless": "^3.37.0",
+    "serverless": "^3.38.0",
     "serverless-plugin-split-stacks": "^1.13.0",
     "shelljs": "^0.8.5",
     "ts-node": "^10.9.1",

--- a/src/index.ts
+++ b/src/index.ts
@@ -15,7 +15,6 @@ import Logging from "./logging";
 import {loadConfig} from "@aws-sdk/node-config-provider";
 import {NODE_REGION_CONFIG_FILE_OPTIONS, NODE_REGION_CONFIG_OPTIONS} from "@aws-sdk/config-resolver";
 import {ChangeAction} from "@aws-sdk/client-route-53";
-import {log} from "util";
 
 class ServerlessCustomDomain {
 

--- a/src/index.ts
+++ b/src/index.ts
@@ -40,7 +40,7 @@ class ServerlessCustomDomain {
         this.options = options;
         Globals.options = options;
 
-        if (v3Utils) {
+        if (v3Utils && v3Utils.log) {
             Globals.v3Utils = v3Utils;
         }
 

--- a/src/index.ts
+++ b/src/index.ts
@@ -15,6 +15,7 @@ import Logging from "./logging";
 import {loadConfig} from "@aws-sdk/node-config-provider";
 import {NODE_REGION_CONFIG_FILE_OPTIONS, NODE_REGION_CONFIG_OPTIONS} from "@aws-sdk/config-resolver";
 import {ChangeAction} from "@aws-sdk/client-route-53";
+import {log} from "util";
 
 class ServerlessCustomDomain {
 
@@ -40,7 +41,7 @@ class ServerlessCustomDomain {
         this.options = options;
         Globals.options = options;
 
-        if (v3Utils && v3Utils.log) {
+        if (v3Utils?.log) {
             Globals.v3Utils = v3Utils;
         }
 

